pymongo/asynchronous/mongo_client.py

```diff
@@ -1355,13 +1355,18 @@ def _close_cursor_soon(
     def _start_session(self, implicit: bool, **kwargs: Any) -> AsyncClientSession:
         server_session = _EmptyServerSession()
         opts = client_session.SessionOptions(**kwargs)
-        return client_session.AsyncClientSession(self, server_session, opts, implicit)
+        bind = opts._bind
+        session = client_session.AsyncClientSession(self, server_session, opts, implicit)
+        if bind:
+            _SESSION.set(session)
```

Comment:
This needs to be removed. We should only bind in `__enter__`.

Reply:
Done in 3c68a70970e96917b0c418900203928bd7678a00

Reply:
@ShaneHarvey OK to resolve this one?